cas-server-support-osf/src/main/java/io/cos/cas/adaptors/postgres/handlers/OpenScienceFrameworkAuthenticationHandler.java

```diff
@@ -134,14 +143,15 @@ protected final HandlerResult authenticateInternal(final OpenScienceFrameworkCre
         }
 
         Boolean validPassphrase = Boolean.FALSE;
+        final String userStatus = verifyUserStatus(user);
 
         if (credential.isRemotePrincipal()) {
             // verified through remote principals
             validPassphrase = Boolean.TRUE;
         } else if (verificationKey != null && verificationKey.equals(user.getVerificationKey())) {
             // verified by verification key
             validPassphrase = Boolean.TRUE;
-        } else if (verifyPassword(plainTextPassword, user.getPassword())) {
+        } else if (plainTextPassword != null && verifyPassword(plainTextPassword, user.getPassword())) {
```

Comment:
The is no need to check `null` for `user.getPassword()` since it can never be null, which is enforced by database constraints.